Tests/TensorFlowTests/OperatorTests/MathTests.swift

```diff
@@ -58,6 +58,78 @@ final class MathOperatorTests: XCTestCase {
         testElementaryFunction(name: "root",
                                { x in root(x, 3) }, { x in Float.root(x, 3) })
     }
+    
+    func testAbs() {
+        let x = Tensor<Float>([-1.0])
+        let y = abs(x)
+        let expectedY = Tensor<Float>([1.0])
+        XCTAssertEqual(y, expectedY)
+    }
+    
+    func testSquaredDifference() {
+        let x = Tensor<Float>([-5.8])
+        let y = Tensor<Float>([5.7])
+        let z = squaredDifference(x, y)
+        let approxZ = Tensor<Float>([132.25])
+        XCTAssertEqual(z, approxZ)
+    }
+    
+    func testZeros() {
+        let x = Tensor<Float>(zeros: [1])
+        let x1 = Tensor<Float>([0.0])
+        XCTAssertEqual(x, x1)
+    }
+    
+    func testLogSoftmax() {
```

Comment:
Given that softmax should only be performed along the last axis, do you want to also add a higher-dimensional test case?

Reply:
Will do!

Reply:
For the softmax implentation, do you mean higher degree tensor input?

Reply:
Yes, exactly. :)